### PR TITLE
[NativeAOT] Remove unused native memcopy helpers.

### PR DESCRIPTION
**Pull request by VSadov**
Managed memcopy nowdays calls `RhBulkMoveWithWriteBarrier` and only when the copy needs to set cards.

The native helpers to do memory copying are dead code and it looks like they have been dead code for a while and may not even function correctly. One example - it looks like EH knew how to handle AV in these, but the support was somehow lost with time.

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/GCMemoryHelpers.cpp [hunk 1]
```diff
@@ -21,91 +21,17 @@
...
-    {
-        pv =
-#if (POINTER_SIZE == 8)
-            bv << 7*8 | bv << 6*8 | bv << 5*8 | bv << 4*8 |
-#endif
-            bv << 3*8 | bv << 2*8 | bv << 1*8 | bv;
-    }
-
-    InlineGCSafeFillMemory(mem, size, pv);
+    InlineGCSafeFillMemory(mem, size, 0);
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/GCMemoryHelpers.cpp:29]: Rename this to `InlineGCSafeZeroMemory` and delete the extra argument?

File: src/coreclr/nativeaot/Runtime/GCMemoryHelpers.cpp [hunk 2]
```diff
@@ -21,91 +21,17 @@
 //
 // USAGE:  The caller is responsible for hoisting any null reference exceptions to a place where the hardware exception
 //         can be properly translated to a managed exception.
-COOP_PINVOKE_CDECL_HELPER(void *, RhpInitMultibyte, (void * mem, int c, size_t size))
+COOP_PINVOKE_CDECL_HELPER(void *, RhpInitMultibyte, (void * mem, size_t size))
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/GCMemoryHelpers.cpp:24]: Change the comment and/or the name to indicate that the buffer can contain GC references?

**Review by jkotas — APPROVED** [review]
LGTM otherwise

File: src/coreclr/nativeaot/Runtime/GCMemoryHelpers.cpp (hunk 2 above)
Comment (VSadov) [reply src/coreclr/nativeaot/Runtime/GCMemoryHelpers.cpp:24]: Right. I am also tempted to move this to a managed helper as well, but I do not see a lot of reasons.
This is only used when unboxing a nullable null.

Comment (VSadov) [reply src/coreclr/nativeaot/Runtime/GCMemoryHelpers.cpp]: I'll move this to managed when/if we have a caller that could realistically need to zero something larger than `BulkMoveWithWriteBarrierChunk`.

**Comment by VSadov** [comment]
Thanks!!

**Pull request merged**